src/platforms/macos/macosutils.mm

```diff
@@ -42,11 +44,35 @@
   Q_ASSERT(appId);
 
   NSString* loginItemAppId =
-      QString("%1.login-item").arg(QString::fromNSString(appId)).toNSString();
-  CFStringRef cfs = (__bridge CFStringRef)loginItemAppId;
-
-  Boolean ok = SMLoginItemSetEnabled(cfs, startAtBoot ? YES : NO);
-  logger.debug() << "Result: " << ok;
+    QString("%1.login-item").arg(QString::fromNSString(appId)).toNSString();
+  
+  // For macOS 13 and beyond, register() and unregister() methods
+  // are used for managing login items since SMLoginItemSetEnabled() is deprecated.
+  // For versions prior to macOS 13, SMLoginItemSetEnabled() is used.
+  #if defined(__MAC_13_0) && (__MAC_OS_X_VERSION_MAX_ALLOWED >= __MAC_13_0)
+    if (@available(macOS 13, *)) {
+      // Use register() or unregister() based on the startAtBoot flag
+      NSError *error = nil;
+      if (startAtBoot) {
+        if (![[SMAppService mainAppService] registerAndReturnError:&error]) {
+          logger.error() << "Failed to register Mozilla VPN LoginItem: " << error.localizedDescription;
+        } else {
+          logger.debug() << "Mozilla VPN LoginItem registered successfully.";
+        }
+      } else {
+        if (![[SMAppService mainAppService] unregisterAndReturnError:&error]) {
+          logger.error() << "Failed to unregister Mozilla VPN LoginItem: " << error.localizedDescription;
+        } else {
+          logger.debug() << "LoginItem unregistered successfully.";
+        }
+      }
+    } else
```

Comment:
Did you consider using something like?
```
#if defined
...
#else
...
#endif
```
I think that's the pattern we mostly use elsewhere in the code.